src/ppc.c

```diff
@@ -0,0 +1,120 @@
+/* cpuid2cpuflags
+ * (c) 2015-2017 Michał Górny
+ * (c) 2019 Georgy Yakovlev
+ * (c) 2019 Shawn Anastasio
+ * 2-clause BSD licensed
+ */
+
+#ifdef HAVE_CONFIG_H
+#	include "config.h"
+#endif
+#include "platforms.h"
+
+#ifdef CPUID_PPC
+
+#include <stdio.h>
+#include <string.h>
+#ifdef HAVE_STDINT_H
+#	include <stdint.h>
+#endif
+#include <assert.h>
+
+#ifdef HAVE_SYS_AUXV_H
+#	include <sys/auxv.h>
+#endif
+#include <sys/utsname.h>
+
+#ifndef __linux__
+#	error "Platform not supported (only Linux supported at the moment)"
+#endif
+#ifndef HAVE_GETAUXVAL
+#	error "Platform not supported (no getauxval())"
+#endif
+
+enum check_type
+{
+	CHECK_SENTINEL = 0,
+	CHECK_HWCAP,
+	CHECK_HWCAP2,
+	CHECK_MAX
+};
+
+struct check_desc
+{
+	enum check_type type;
+	unsigned long mask;
+};
+
+struct flag_info
+{
+	const char* name;
+	struct check_desc checks[3];
+};
+
+struct flag_info flags[] = {
+	/* taken from /usr/include/bits/hwcap.h */
+	/* PPC_FEATURE_HAS_ALTIVEC */
+	{ "altivec", {{ CHECK_HWCAP, 0x10000000 }} },
+	/* PPC_FEATURE_HAS_VSX */
+	{ "vsx", {{ CHECK_HWCAP, 0x00000080 }} },
+	/* only ISA 3.00 CPU is P9, but it's possible to build a compliant
+	 * CPU that only implements the scalar instructions and not vsx */
+	/*  PPC_FEATURE_HAS_VSX | PPC_FEATURE2_ARCH_3_00 */
+	{ "vsx3", {{ CHECK_HWCAP, 0x00000080 }, { CHECK_HWCAP2, 0x00800000 }} },
```

Comment:
I'm confused by this. The comment is very roundabout but it sounds like you want vsx3 only if both conditions are true, while you print it if *either* is true.